carma_wm/src/collision_detection.cpp

```diff
@@ -25,17 +26,20 @@ namespace carma_wm {
 
         collision_detection::MovingObject ConvertRoadwayObstacleToMovingObject(const cav_msgs::RoadwayObstacle& rwo){
 
-            collision_detection::MovingObject ro;
+            collision_detection::MovingObject mo;
             
-            ro.object_polygon = ObjectToBoostPolygon<polygon_t>(rwo.object.pose.pose, rwo.object.size);
+            mo.object_polygon = ObjectToBoostPolygon<polygon_t>(rwo.object.pose.pose, rwo.object.size);
 
+            // Add future polygons for roadway obstacle
             for (auto i : rwo.object.predictions){
-                ro.future_polygons.push_back(ObjectToBoostPolygon<polygon_t>(i.predicted_position, rwo.object.size));
+                std::tuple <__uint32_t,polygon_t> future_object(i.header.stamp.nsec/1000,ObjectToBoostPolygon<polygon_t>(i.predicted_position, rwo.object.size));
```

Comment:
I'm fairly certain the .nsec field is only the number of nsec past the current sec mark. So this will generate errors when the seconds roll over. You will need to combine the seconds value for an accurate absolute reference time. 

In addition you can't use a uint32_t to store microseconds since the current epoch of 1970 since the uint32 has a max value of 4,294,967,295
and the current unix time is 1.594228e+15 microseconds. 

If you want to use a singular time value I would suggest either using milliseconds or taking the first time stamp and subtracting it from all others before doing the conversion to micro seconds. If you use milliseconds please document that this is the max resolution.